lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,20 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      paths = Set.new(applicable_files.map { |file| File.dirname(file) + File::SEPARATOR })
+      repo_files = Set.new(Overcommit::GitRepo.list_files(paths.to_a) + applicable_files)
```

Comment:
Ah, I didn't realize. In hindsight this makes sense since a set doesn't have the concept of order of elements.